services/workflows-service/src/business-report/business-report.controller.external.ts

```diff
@@ -111,13 +111,24 @@ export class BusinessReportControllerExternal {
     let business: Pick<Business, 'id' | 'correlationId'> | undefined;
     const merchantNameWithDefault = merchantName || 'Not detected';
 
-    if (!businessCorrelationId) {
+    if (businessCorrelationId) {
```

Comment:
**Before:**
1. If a correlation ID is present, use it to fetch the existing business.
2. Otherwise, create a new business.

**After:**
1. If a correlation ID is present, fetch the business if it exists.
2. Otherwise (if the correlation ID is missing or the business doesn't exist), create a new business and associate the correlation ID with it, if available.